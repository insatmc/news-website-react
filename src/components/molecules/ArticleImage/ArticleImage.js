import React from "react"
import PropTypes from "prop-types"
import { useTranslation } from "react-i18next"

import Text from "../../atoms/Text/Text"

import "./ArticleImage.scss"

export default function ArticleImage({ image, title, onRemove }) {
  const { t } = useTranslation()

  const handleRemove = (e) => {
    e.stopPropagation()
    e.preventDefault()
    onRemove()
  }

  return (
    <div className="articleImage" style={{ backgroundImage: `url(${image})` }}>
      <button
        data-testid="button-remove"
        type="button"
        className="remove-btn"
        onClick={handleRemove}
      >
        {t("Archive")}
      </button>
      <Text size="l" color="#FFF" kind="bold">
        {title}
      </Text>
    </div>
  )
}

ArticleImage.propTypes = {
  image: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  onRemove: PropTypes.func.isRequired,
}
