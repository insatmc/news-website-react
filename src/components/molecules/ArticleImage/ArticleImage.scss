.articleImage {
    height: 200px;
    background-size: cover;
    display: flex;
    padding: 20px;
    align-items: flex-end;
    position: relative;
    background-color: #b2bec3;

    &::before {
        content: ' ';
        background: linear-gradient(0deg, rgba(0,0,0,0.6) 0%, rgba(0,212,255,0) 100%);
        position: absolute;
        top: 0;
        left: 0;
        width: 100%;
        height: 100%;
        z-index: 0;
    }

    > * {
        z-index: 1;
    }

    .remove-btn{
        position: absolute;
        top: 8px;
        right: 8px;
        border: none;
        background: white;
        height: 16px;
        padding: 0 8px;
        display: flex;
        justify-content: center;
        align-items: center;

        cursor: pointer;

        &:hover {
            background: #c0392b;
            color: white;
        }
    }
}