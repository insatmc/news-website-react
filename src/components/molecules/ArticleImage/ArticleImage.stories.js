import React from "react"
import ArticleImage from "./ArticleImage"

export const Default = () => (
  <ArticleImage
    title="Title of article"
    image="https://lh3.googleusercontent.com/w5zAdJ2zDj8ActvZEVESIqL0pevo1s4B7uzLhjn8bc9LRqPo5QM78OzLyCqjcDNYa6w"
  />
)

export default {
  title: "Components/Molecules/ArticleImage",
  component: ArticleImage,
}
