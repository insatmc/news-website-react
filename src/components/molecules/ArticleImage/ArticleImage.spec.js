import React from "react"
import { render, fireEvent } from "@testing-library/react"
import ArticleImage from "./ArticleImage"

describe("Article Image", () => {
  it("should match snapshot", () => {
    const { wrapper } = render(
      <ArticleImage
        title="Title of article"
        image="https://lh3.googleusercontent.com/w5zAdJ2zDj8ActvZEVESIqL0pevo1s4B7uzLhjn8bc9LRqPo5QM78OzLyCqjcDNYa6w"
        onRemove={jest.fn()}
      />
    )

    expect(wrapper).toMatchSnapshot()
  })

  it("should call onRemove", () => {
    const onRemove = jest.fn()
    const { getByTestId } = render(
      <ArticleImage
        title="Title of article"
        image="https://lh3.googleusercontent.com/w5zAdJ2zDj8ActvZEVESIqL0pevo1s4B7uzLhjn8bc9LRqPo5QM78OzLyCqjcDNYa6w"
        onRemove={onRemove}
      />
    )

    fireEvent.click(getByTestId("button-remove"))

    expect(onRemove).toHaveBeenCalled()
  })
})
