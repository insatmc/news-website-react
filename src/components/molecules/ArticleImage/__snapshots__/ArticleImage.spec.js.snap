// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article Image should match snapshot 1`] = `undefined`;
